be/src/vec/core/block.cpp

```diff
@@ -863,6 +876,10 @@ void MutableBlock::add_row(const Block* block, int row) {
 }
 
 void MutableBlock::add_rows(const Block* block, const int* row_begin, const int* row_end) {
+    if (_type == BlockType::DYNAMIC) {
```

Comment:
It may be more efficient to add UNLIKELY since most tables are without dynamic schema.